File: Rentify.Services/Database/Property.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Rentify.Services.Database
{
    public class Property
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(UserId))]
        public int UserId { get; set; }
        public User User { get; set; }

        public string Name { get; set; }
        public string Location { get; set; }

        [ForeignKey(nameof(CityId))]
        public int CityId { get; set; }
        public City City { get; set; }

        [ForeignKey(nameof(BuildingTypeId))]
        public int BuildingTypeId { get; set; }
        public BuildingType BuildingType { get; set; }

        public decimal PricePerDay { get; set; }
        public decimal PricePerMonth { get; set; }
        public List<string> Tags { get; set; }
        public double SquareMeters { get; set; }
        public string Details { get; set; }
        public bool IsRentingPerDay { get; set; }
        public bool IsActiveOnApp { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
    }
}

File: UI/rentify_mobile/lib/models/property.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rentify_mobile/models/city.dart';
import 'package:rentify_mobile/models/building_type.dart';
import 'package:rentify_mobile/models/user.dart';

part 'property.g.dart';

@JsonSerializable()
class Property {
  final int id;
  final int userId;
  User? user;
  final String name;
  final String location;
  final int cityId;
  City? city;
  final int buildingTypeId;
  BuildingType? buildingType;
  final double pricePerDay;
  final double pricePerMonth;
  final double squareMeters;
  List<String>? tags;
  final String details;
  final bool isAvailable;
  final bool isRentingPerDay;
  final bool isActiveOnApp;
  final String? whyRecommended;
  final double? latitude;
  final double? longitude;

  Property({
    required this.id,
    required this.userId,
    this.user,
    required this.name,
    required this.location,
    required this.cityId,
    this.city,
    required this.buildingTypeId,
    this.buildingType,
    required this.pricePerDay,
    required this.pricePerMonth,
    required this.squareMeters,
    this.tags,
    required this.details,
    required this.isAvailable,
    required this.isRentingPerDay,
    required this.isActiveOnApp,
    this.whyRecommended,
    this.latitude,
    this.longitude,
  });

  factory Property.fromJson(Map<String, dynamic> json) =>
      _$PropertyFromJson(json);

  Map<String, dynamic> toJson() => _$PropertyToJson(this);
}
